backend/dataall/modules/s3_datasets/services/dataset_service.py

```diff
@@ -56,6 +56,16 @@ class DatasetService:
     def register(cls, interface: DatasetServiceInterface):
         cls._interfaces.append(interface)
 
+    @classmethod
+    def _resolve_consumption_details_resource_uri(cls, session, uri, group_uri) -> list:
+        """Find resources from other modules that need to be accessed"""
+        for interface in cls._interfaces:
+            consumption_detail = interface.find_consumption_details_resource_uri(session, uri, group_uri)
```

Comment:
`find_consumption_details_resource_uri` is not part of the DatasetServiceInteface abstract classes, therefore when adding new interfaces it might fail because the method is not mandatory to be implemented

Reply:
yup - `find_resources_by_uri` from DatasetServiceInteface to be renamed `find_consumption_details_resource_uri` 

... but may be removed if removing Preview ability fro confidential datasets

Reply:
removed as part of removing Preview Tab for Confidential Data